src/libcore/array.rs

```diff
@@ -220,25 +220,43 @@ array_impls! {
     30 31 32
 }
 
-// The Default impls cannot be generated using the array_impls! macro because
+// Some impls cannot be generated using the array_impls! macro because
 // they require array literals.
 
-macro_rules! array_impl_default {
-    {$n:expr, $t:ident $($ts:ident)*} => {
+macro_rules! array_impl_variadic {
+    {$n:expr, $i:ident: $t:ident $(, $is:ident: $ts:ident)*} => {
         #[stable(since = "1.4.0", feature = "array_default")]
         impl<T> Default for [T; $n] where T: Default {
             fn default() -> [T; $n] {
                 [$t::default(), $($ts::default()),*]
             }
         }
-        array_impl_default!{($n - 1), $($ts)*}
+
+        #[unstable(feature = "array_from_tuple", reason = "recently added", issue = "0")]
```

Comment:
These'll be insta-stable unfortunately so could these be `#[stable]` instead of `#[unstable]`?